### PR TITLE
[BPK-2460] Add Horizontal Navigation tests

### PR DESCRIPTION
**Pull request by georgegillams**
+ [x] Check this if you have read and followed the [contributing guidelines](https://github.com/Skyscanner/backpack-ios/blob/master/CONTRIBUTING.md)

Remember to include the following changes:
+ [x] `UNRELEASED.md`
+ [x] `README.md`
+ [x] Tests
+ [x] Docs in this PR https://github.com/Skyscanner/backpack-docs/pull/73

_If you are curious about how we review, please read through the [code review guidelines](https://github.com/Skyscanner/backpack/blob/master/CODE_REVIEW_GUIDELINES.md)_

### CODE REVIEW

File: Backpack/HorizontalNavigation/README.md [hunk 1]
```diff
@@ -1,3 +1,40 @@
...
+```swift
+import Backpack
+
+let options = [
+    BPKHorizontalNavigationOption(name: "Flights", iconName: .flight),
+    BPKHorizontalNavigationOption(name: "Hotels", iconName: .hotels),
+    BPKHorizontalNavigationOption(name: "Car hire", iconName: .cars)
+]
+let horizontalNavigation = Backpack.HorizontalNavigation(options: options, selected:0)
+horizontalNavigation.subview = myInnerView
```
Comment (k0nserv) [inline Backpack/HorizontalNavigation/README.md]: What even is this line?

File: Example/Backpack NativeUITests/HorizontalNavigationUITest.swift [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+        attachment.name = name
+        add(attachment)
+    }
+
+    func navigateAndShow() {
+        app.tables.staticTexts["Horizontal navigation"].tap()
+        app.tables.staticTexts["Default"].tap()
+    }
+
+    func testTapChipToSelect() {
```
Comment (k0nserv) [inline Example/Backpack NativeUITests/HorizontalNavigationUITest.swift]: This function name is outdated

File: Example/Backpack.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -584,7 +592,11 @@
 		D6D81D60213FEEB90013BB96 /* SnapshotTests */ = {
 			isa = PBXGroup;
 			children = (
+<<<<<<< HEAD
```
Comment (k0nserv) [inline Example/Backpack.xcodeproj/project.pbxproj]: This has a conflict

File: Example/Tests/BPKHorizontalNavigationTest.m [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+- (void)testSetOptions {
+    BPKHorizontalNavigation *horizontalNavigation = [[BPKHorizontalNavigation alloc] initWithOptions:@[
+                                                                                                       [[BPKHorizontalNavigationOption alloc] initWithName:@"Flights"],
+                                                                                                       [[BPKHorizontalNavigationOption alloc] initWithName:@"Hotels"],
+                                                                                                       [[BPKHorizontalNavigationOption alloc] initWithName:@"Car hire"]
+                                                                                                       ] selected:0];
+
+    horizontalNavigation.selectedItemIndex = 1;
+    horizontalNavigation.options = @[
+    [[BPKHorizontalNavigationOption alloc] initWithName:@"Flights 2"],
```
Comment (k0nserv) [inline Example/Tests/BPKHorizontalNavigationTest.m]: Can we indent this properly?

File: Backpack/HorizontalNavigation/README.md (hunk 1 above)
Comment (georgegillams) [reply Backpack/HorizontalNavigation/README.md]: I don't even know 😂

**Pull request merged**